ParticleObjects/Source/pointSource_class.f90

```diff
@@ -131,19 +136,30 @@ subroutine init(self, dict, geom)
 
     ! Get particle energy/group
     isCE = dict % isPresent('E')
-    isMG = dict % isPresent('G')
+    isMG = dict % isPresent('probG')
     if (isCE .and. isMG) then
       call fatalError(Here, 'Source may be either continuous energy or MG, not both')
 
     elseif (isCE) then
       call dict % get(self % E, 'E')
 
     elseif (isMG) then
-      call dict % get(self % G, 'G')
+      call dict % get(temp, 'probG')
+      nucData => ndReg_getNeutronMG()
+      if(.not.associated(nucData)) call fatalError(Here, 'Failed to retrieve Nuclear Database')
+
+      if (size(temp) /= nucData % nG) then
+        call fatalError(Here, 'Source energy group distribution must have '//numToChar(nucData % nG)//' groups')
+      end if
+
+      allocate(self % probG(nucData % nG))
+      self % probG = temp
+      self % probG = self % probG / self % probG
```

Comment:
Should it be divided by the sum?

Reply:
Thank you for spotting that. I could swear I did it... Need to write unit tests for sources :/